saleor/graphql/meta/mutations.py

```diff
@@ -0,0 +1,185 @@
+    value = graphene.String(required=True, description="Stored metadata value.")
+
+
+class UpdateMetadata(BaseMetadataMutation):
+    class Meta:
+        description = "Updates metadata for item."
+        permission_map = PUBLIC_META_PERMISSION_MAP
+        error_type_class = MetadataError
+        error_type_field = "metadata_errors"
+
+    class Arguments:
+        id = graphene.ID(description="ID of an object to update.", required=True)
+        input = MetadataItemInput(
+            description="Fields required to update new or stored metadata item.",
+            required=True,
+        )
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        instance = cls.get_instance(info, **data)
+        if instance:
+            metadata = data.pop("input")
+            item = {metadata.key: metadata.value}
+            instance.store_meta(items=item)
+            instance.save(update_fields=["meta"])
+        return cls.success_response(instance)
+
+
+class DeleteMetadata(BaseMetadataMutation):
+    class Meta:
+        description = "Delete metadata for key."
+        permission_map = PUBLIC_META_PERMISSION_MAP
+        error_type_class = MetadataError
+        error_type_field = "metadata_errors"
+
+    class Arguments:
+        id = graphene.ID(description="ID of an object to update.", required=True)
+        key = graphene.String(description="Key for value to delete.", required=True)
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        instance = cls.get_instance(info, **data)
+        if instance:
+            metadata_key = data.pop("key")
+            instance.delete_meta(metadata_key)
+            instance.save(update_fields=["meta"])
+        return cls.success_response(instance)
+
+
+class UpdatePrivateMetadata(BaseMetadataMutation):
+    class Meta:
+        description = "Updates private metadata for item."
+        permission_map = PRIVATE_META_PERMISSION_MAP
+        error_type_class = MetadataError
+        error_type_field = "metadata_errors"
+
+    class Arguments:
+        id = graphene.ID(description="ID of an object to update.", required=True)
+        input = MetadataItemInput(
+            description=(
+                "Fields required to update new or stored private metadata item."
+            ),
+            required=True,
+        )
+
+    @classmethod
+    def perform_mutation(cls, root, info, **data):
+        instance = cls.get_instance(info, **data)
+        if instance:
+            metadata = data.pop("input")
+            item = {metadata.key: metadata.value}
+            instance.store_private_meta(items=item)
+            instance.save(update_fields=["private_meta"])
+        return cls.success_response(instance)
+
+
+class DeletePrivateMetadata(BaseMetadataMutation):
+    class Meta:
+        description = "Delete private metadata for key."
+        permission_map = PRIVATE_META_PERMISSION_MAP
+        error_type_class = MetadataError
```

Comment:
But in witch place, because I can't set it inside `class Meta` in `BaseMetadataMutation`.
Abstract mutation in graphene couldn't have any field inside `class Meta`.